### PR TITLE
Feature/dockerfiles

### PR DESCRIPTION
**Pull request by jackmonty**
- use per stage agent
- added docker-package stage to result and vote app's
- updated test steps for result and vote app's

** be extra careful as this can not be tested with CI Server due to master branch conditional **

### CODE REVIEW

File: result/Jenkinsfile [hunk 1]
```diff
@@ -18,15 +19,38 @@ pipeline {
...
 				changeset "**/result/**"
 			}
             steps {
                 echo 'Running Unit Test on result app'
 				dir('result') {
+					sh 'npm install'
 					sh 'npm test'
 				}
             }
+        }
```
Comment (jackmontyorg) [inline result/Jenkinsfile:37]: additional curly brace here, needs to be deleted

File: vote/Jenkinsfile [hunk 2]
```diff
@@ -19,15 +20,39 @@ pipeline {
...
 				changeset "**/vote/**"
 			}
             steps {
                 echo 'Running Unit Test on vote app'
 				dir('vote') {
+					sh 'pip install -r requirements.txt'
 					sh 'nosetests -v'
 				}
             }
+        }
```
Comment (jackmontyorg) [inline vote/Jenkinsfile:39]: double check that this is in the correct place.

**Review by jackmontyorg — CHANGES_REQUESTED** [review]
both Jenkinsfiles need to be checked to ensure matched brackets are correct.

File: result/Jenkinsfile (hunk 1 above)
Comment (jackmonty) [reply result/Jenkinsfile:37]: checked and okay

File: vote/Jenkinsfile (hunk 2 above)
Comment (jackmonty) [reply vote/Jenkinsfile:39]: checked and okay

**Comment by jackmonty** [comment]
both Jenkinsfiles checked and confirmed correct.

**Review by jackmontyorg — APPROVED** [review]
all ok after a check, thanks!

**Pull request merged**